plaid-stl/src/aws/kms.rs

```diff
@@ -0,0 +1,193 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+use crate::PlaidFunctionError;
+
+const RETURN_BUFFER_SIZE: usize = 1024 * 1024 * 4; // 4 MiB
+
+/// Represents the response from a sign request, including the key ID, signature, and signing algorithm used.
+#[derive(Deserialize)]
+pub struct SignRequestResponse {
+    /// The Amazon Resource Name (key ARN) of the asymmetric KMS key that was used to sign the message.
+    pub key_id: Option<String>,
+    /// The cryptographic signature that was generated for the message.
+    ///
+    /// - When used with the supported RSA signing algorithms, the encoding of this value is defined by PKCS #1 in [RFC 8017](https://tools.ietf.org/html/rfc8017).
+    /// - When used with the `ECDSA_SHA_256`, `ECDSA_SHA_384`, or `ECDSA_SHA_512` signing algorithms, this value is a DER-encoded object as defined by ANSI X9.62–2005 and [RFC 3279 Section 2.2.3](https://tools.ietf.org/html/rfc3279#section-2.2.3). This is the most commonly used signature format and is appropriate for most uses.
+    ///
+    /// When you use the HTTP API or the Amazon Web Services CLI, the value is Base64-encoded. Otherwise, it is not Base64-encoded.
+    pub signature: Option<Vec<u8>>,
+    /// The signing algorithm that was used to sign the message.
+    pub signing_algorithm: Option<String>,
+}
+
+/// Contains metadata about a KMS key
+#[derive(Deserialize)]
+pub struct GetPublicKeyResponse {
+    /// The Amazon Resource Name (key ARN) of the asymmetric KMS key that was used to sign the message.
+    pub key_id: Option<String>,
+    /// The exported public key.
+    ///
+    /// The value is a DER-encoded X.509 public key, also known as SubjectPublicKeyInfo (SPKI), as defined in RFC 5280.
+    /// When you use the HTTP API or the Amazon Web Services CLI, the value is Base64-encoded. Otherwise, it is not Base64-encoded.
+    pub public_key: Option<Vec<u8>>,
+}
+
+/// Tells KMS whether the value of the Message parameter should be hashed as part of the signing algorithm.
+/// Use `Raw` for unhashed messages; use `Digest` for message digests, which are already hashed.
+#[derive(Serialize)]
+pub enum MessageType {
+    Raw,
+    Digest,
+}
+
+/// The signing algorithm to use in signing this request
+pub enum SigningAlgorithm {
```

Comment:
Is there nothing that distinguishes the K1 and R1 curves?

Reply:
Not that I'm aware of - here is the AWS docs:
https://docs.rs/aws-sdk-kms/1.42.0/aws_sdk_kms/types/enum.SigningAlgorithmSpec.html